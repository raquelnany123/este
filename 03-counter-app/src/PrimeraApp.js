import React from "react";
import PropTypes from 'prop-types';
// import { Fragment } from "react/cjs/react.production.min";

// este saludo es una desestructuracion de los props que estan en el index
const PrimeraApp = ({ saludo, subtitulo}) => {


return(
    <>
    <h1> {saludo } </h1>
    <p> {subtitulo} </p>
    </>
)


};

PrimeraApp.propType={
    // es que espera el saludo d tipo string y que sea obligatorio
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {

    subtitulo: 'soy un subtitulo'
}


export default PrimeraApp;
