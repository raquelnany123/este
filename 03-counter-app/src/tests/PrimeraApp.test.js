import React from 'react';
import { shallow } from 'enzyme'
import PrimeraApp from "../PrimeraApp";


describe('Pruebas en <PrimeraAPP />>', () => {

    // test(' debe de estar el mensaje "Hola, Soy Goku" ', () => {
        
    //     const saludo = 'Hola, soy Goku';


    //     // wrapper = producto rendemizado de lo que se quiere mostrar
    //     const {getByText} = render( <PrimeraApp saludo = { saludo } />)

    //     expect( getByText(saludo) ).toBeInTheDocument();

    // });

    test('debe de mostrar <PrimeraApp /> correctamente', () =>{

        const saludo = 'Hola, Soy goku';
        const wrapper = shallow( <PrimeraApp />)
    
        expect( wrapper ).toMatchSnapshot();

    });
    

    
})
