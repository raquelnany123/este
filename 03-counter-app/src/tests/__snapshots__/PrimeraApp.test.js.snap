// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraAPP />> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
     
  </h1>
  <p>
     
    soy un subtitulo
     
  </p>
</Fragment>
`;
